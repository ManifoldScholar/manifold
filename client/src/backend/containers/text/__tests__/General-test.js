import TextGeneralContainer from "../General";

describe("backend/containers/text/General", () => {
  def("creator", () => factory("maker"));
  def("contributor", () => factory("maker"));
  def("text", () =>
    factory("text", {
      relationships: { creators: [$creator], contributors: [$contributor] }
    })
  );
  def("root", () => <TextGeneralContainer text={$text} />);

  it("matches the snapshot when rendered", () => {
    expect(mount($withApp($root)).html()).toMatchSnapshot();
  });
});
