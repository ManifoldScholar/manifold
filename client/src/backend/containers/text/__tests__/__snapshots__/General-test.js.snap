// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`backend/containers/text/General matches the snapshot when rendered 1`] = `"<section><div><form class=\\"form-secondary\\" data-id=\\"submit\\"><div class=\\"form-section form-section--primary\\" role=\\"group\\" aria-labelledby=\\"1-header\\" aria-describedby=\\"1-instructions\\"><header class=\\"form-section-label\\"><h2 id=\\"1-header\\">General</h2></header><div class=\\"form-input-group form-input-group--primary\\"><div class=\\"form-input wide\\"><label for=\\"text-input-1\\" class=\\"\\">Title</label><input id=\\"text-input-1\\" type=\\"text\\" placeholder=\\"Enter Text Title\\" aria-describedby=\\"text-input-error-1 text-input-instructions-1\\" value=\\"Ain't No Thang\\"></div><div class=\\"form-input wide\\"><label for=\\"text-input-1\\" class=\\"\\">Subtitle</label><input id=\\"text-input-1\\" type=\\"text\\" placeholder=\\"Enter Subtitle\\" aria-describedby=\\"text-input-error-1 text-input-instructions-1\\" value=\\"\\"></div><div class=\\"form-input form-date-picker\\"><h4 class=\\"form-input-heading\\">Publication Date</h4><div class=\\"react-datepicker-wrapper\\"><div class=\\"react-datepicker__input-container\\"><div class=\\"form-input\\"><label for=\\"masked-text-1\\" class=\\"\\"></label><react-text-mask value=\\"12/04/2001\\" id=\\"masked-text-1\\" type=\\"text\\" mask=\\"/\\\\d/,/\\\\d/,/,/\\\\d/,/\\\\d/,/,/\\\\d/,/\\\\d/,/\\\\d/,/\\\\d/\\" placeholderchar=\\"_\\" aria-describedby=\\"masked-text-instructions-1\\"></react-text-mask></div></div></div><button type=\\"button\\" class=\\"form-date-picker__button\\">Clear</button></div><div class=\\"form-input wide\\"><label for=\\"text-input-1\\" class=\\"\\">Slug</label><input id=\\"text-input-1\\" type=\\"text\\" placeholder=\\"Enter Text Slug\\" aria-describedby=\\"text-input-error-1 text-input-instructions-1\\" value=\\"\\"></div><div class=\\"form-input wide\\"><div class=\\"form-input\\"><label for=\\"textarea-1\\" class=\\"\\">Description</label><textarea id=\\"textarea-1\\" aria-describedby=\\"textarea-error-1 textarea-instructions-1\\" style=\\"height: 100px;\\" placeholder=\\"Enter a Brief Description\\"></textarea></div></div><div class=\\"form-input wide\\"><div class=\\"form-input\\"><label for=\\"upload-1\\" class=\\"\\">Cover</label><div><input><div class=\\"contents-empty\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"manicon-svg contents-empty__icon svg-icon--upload64\\" width=\\"82\\" height=\\"82\\" viewBox=\\"0 0 64 64\\" aria-hidden=\\"true\\"><g fill=\\"none\\" fill-rule=\\"evenodd\\"><path fill=\\"currentColor\\" fill-rule=\\"nonzero\\" d=\\"M32.9988,29.9144136 L32.9988,42.0002 L30.9988,42.0002 L30.9988,29.9144136 L25.7059068,35.2073068 L24.2916932,33.7930932 L31.9988,26.0859864 L39.7059068,33.7930932 L38.2916932,35.2073068 L32.9988,29.9144136 Z M50.1861877,22.5986557 C56.4497399,23.0677704 61.5405685,28.1667806 61.9686336,34.4041458 C62.4434658,41.2940362 57.365095,47.2462232 50.5614005,47.9604353 L50.4578781,47.9658996 L11.6699429,47.9977815 C8.53033255,47.7930939 5.62013166,45.6673177 3.63424187,42.4514574 C-0.606099659,35.5873073 3.82653587,26.5383733 11.8099093,26.1212576 C14.0599364,26.0038929 16.2272667,26.566062 18.0852466,27.6896237 C18.5412167,20.3542541 24.6631146,14.5469 32.145,14.5469 C37.9174389,14.5469 43.0124004,18.037155 45.1604639,23.2141812 C46.7581064,22.6838525 48.4567018,22.4699946 50.1861877,22.5986557 Z M44.9652102,25.4242014 C44.4268603,25.6530481 43.8078886,25.3729597 43.6244437,24.8174965 C42.0077096,19.9220973 37.4048058,16.5469 32.145,16.5469 C25.4688521,16.5469 20.058,21.9306221 20.058,28.5689 C20.058,28.8683971 20.0747493,29.1712969 20.1115458,29.6128545 C20.1868189,30.5161318 19.1178416,31.0455547 18.4449777,30.438241 C16.6714657,28.837504 14.3554563,27.9911971 11.9141769,28.1185379 C5.4789839,28.4547641 1.87879742,35.8043081 5.33584257,41.4004793 C7.00940449,44.1105699 9.38308844,45.8444436 11.7341219,45.9999004 L50.403086,45.965947 C56.1172051,45.3394064 60.3725258,40.33352 59.9733467,34.5413681 C59.6134776,29.2976912 55.3109878,24.9882098 50.0375627,24.5931257 C48.2705385,24.4617114 46.5473971,24.7516308 44.9652102,25.4242014 Z\\"></path></g></svg><div class=\\"message\\"><p class=\\"primary\\"><span class=\\"form-dropzone__upload-prompt\\">Upload a file</span> or <br>drag and drop here<br></p><p class=\\"secondary\\">gif, jpeg, jpg, png</p></div></div></div></div></div></div></div><div class=\\"form-section form-section--primary\\" role=\\"group\\" aria-labelledby=\\"1-header\\" aria-describedby=\\"1-instructions\\"><header class=\\"form-section-label\\"><h2 id=\\"1-header\\">Presentation</h2></header><div class=\\"form-input-group form-input-group--primary\\"><div class=\\"form-input wide form-toggle-secondary\\"><label for=\\"switch-input-1\\" class=\\"form-switch checked\\"><span class=\\"form-input-heading toggle above\\">Published?</span><input type=\\"checkbox\\" id=\\"switch-input-1\\" checked=\\"\\"><span class=\\"toggle-indicator\\"><span class=\\"boolean-primary checked\\" aria-hidden=\\"true\\"></span></span><span class=\\"instructions\\">When on, a \\"published\\" badge will appear next to the text in lists of texts.</span></label></div><div class=\\"form-input wide\\"><label for=\\"text-input-1\\" class=\\"has-instructions\\">Section Label</label><input id=\\"text-input-1\\" type=\\"text\\" placeholder=\\"Section\\" aria-describedby=\\"text-input-error-1 text-input-instructions-1\\" value=\\"\\"><span class=\\"instructions\\" id=\\"text-input-instructions-1\\">For example, “chapter” for books or “article” for journals</span></div></div></div><div class=\\"form-section form-section--primary\\" role=\\"group\\" aria-labelledby=\\"1-header\\" aria-describedby=\\"1-instructions\\"><header class=\\"form-section-label\\"><h2 id=\\"1-header\\">Access</h2></header><div class=\\"form-input-group form-input-group--primary\\"><div class=\\"form-input wide form-toggle-secondary\\"><label for=\\"switch-input-1\\" class=\\"form-switch\\"><span class=\\"form-input-heading toggle above\\">Ignore Access Restrictions?</span><input type=\\"checkbox\\" id=\\"switch-input-1\\"><span class=\\"toggle-indicator\\"><span class=\\"boolean-primary\\" aria-hidden=\\"true\\"></span></span><span class=\\"instructions\\">When on, this text will be available to all readers regardless of project access settings.</span></label></div></div></div><div class=\\"form-input submit wide\\"><input class=\\"button-secondary button-secondary--outlined\\" type=\\"submit\\" value=\\"Save Text\\"></div></form></div></section>"`;
